base/common/DateLUTImpl.h

```diff
@@ -274,6 +274,23 @@ class DateLUTImpl
         return res / 3600;
     }
 
+
+    inline time_t timezoneOffset(time_t t) const
+    {
+        DayNum index = findIndex(t);
+
+        /// Calculate daylight saving offset first, ignore the leap seconds
+        time_t res = (lut[index].date - lut[0].date) % 86400;
+        res = res > 43200 ? (86400 - res) : (0 - res);
```

Comment:
The calculation seem to be correct, but it is better to add comment about why so.

Reply:
yes, I've appended comments.